### PR TITLE
test: Fix escapeField property-based tests for custom quotations and delimiters

### PR DESCRIPTION
**Pull request by VaishnaviOnPC**
## Description

This PR fixes failing property-based tests in escapeField.spec.ts related to:
* Custom quotation characters – Correctly asserts that internal quotation characters are doubled according to the escapeField logic.
* Custom delimiters – Updates assertions to check for correct escaping and quoting instead of splitting on delimiters (handles quoted delimiters safely).
* All other existing tests remain unchanged. After these updates, all tests for escapeField now pass reliably.

## Changes

* Updated property-based test for custom quotation characters to correctly calculate doubled internal quotations.
* Updated property-based test for custom delimiters to assert proper quoting rather than splitting by delimiters.
* Retained existing functional and edge-case tests for escapeField.

## Testing

* Ran pnpm test escapeField locally.
* Verified that all 7 tests now pass consistently.

## Summary by CodeRabbit

* **Tests**
  * Enhanced test coverage with expanded property-based test scenarios for field escaping functionality, including validation of quotation handling, special character processing, and delimiter management.

### CODE REVIEW

File: src/escapeField.spec.ts [hunk 1]
```diff
@@ -15,3 +16,113 @@ describe("escapeField function", () => {
...
+        }
+      }),
+    );
+  });
+
+  it("should work with custom quotation characters", () => {
+    fc.assert(
+      fc.property(
+        fc.array(fc.string()),
+        fc.constantFrom("'", "`", "$", "cc"),
```
Comment (kamiazya) [inline src/escapeField.spec.ts]: This applies to other parts as well, but rather than using your own definitions, I think it's better to use `FC.text()`, `FC.quotation()`, etc., which are already implemented in `src/__tests__/helper.ts`.

Also, the essence of the test is at the field level, so there's no need to do it at the row level using `fc.array(fc.string())`. I think `fc.property(FC.text(), FC.quotation())` or similar would be better.

**Review by kamiazya — CHANGES_REQUESTED** [review]

**Review by kamiazya — APPROVED** [review]

**Comment by kamiazya** [comment]
@VaishnaviOnPC LGTM, thanks!
